portindex2json/portindex2json.tcl

```diff
@@ -12,6 +12,43 @@
 
 package require json::write
 
+proc format_string {string} {
+	if {[llength $string] > 1} {
+		foreach element $string {
+			lappend new_string "\"$element\""
+		}
+	} else {
+		set new_string [list "\"$string\""]
+	}
+	return $new_string
+}
+
+proc get_maintainer_array {maintainer} {
+	set detail_list [split $maintainer @]
+	if {[string first : [lindex $detail_list 0]] != -1} {
+	    set email_list [split [lindex $detail_list 0] :]
+		set name [string trimright [lindex $email_list 1]]
+		set email(name) "\"$name\""
+		set email(domain) "\"[lindex $email_list 0]\""
+		::json::write indented false
+		set email_subobject [::json::write object {*}[array get email]]
+		::json::write indented true
+	} else {
+		set name [string trimright [lindex $detail_list 0]]
+		set email(name) "\"$name\""
+		set email(domain) "\"macports.org\""
+		::json::write indented false
+		set email_subobject [::json::write object {*}[array get email]]
+		::json::write indented true
+	}
+
+    set maintainer_array(github) "\"[lindex $detail_list 1]\""
+    set maintainer_array(email) $email_subobject
```

Comment:
You seem to be mixing tabs and spaces for indentation. Please use one or the other. The project's preferred style is 4 spaces.

Reply:
My bad, I will fix it. Shall I opt for a separate commit or squash?

Reply:
I have converted all spaces to tabs, but it is automatically getting converted to use 8 spaces.

Reply:
Nothing got converted in the process, the code is still using tabs, but GitHub and most other tools will always show the equivalent of 8 spaces where you were using a tab. GitHub has absolutely no idea how many spaces you configured in your text editor.

While I do have my own preference for the usage of tabs vs. spaces, what I care about most is consistency, and MacPorts is using 4 spaces almost exclusively (graveyard ports make a justified exception), so I don't see a good reason for switching to tabs just in this one file.

Reply:
So the right way to get through is to use 4 spaces in place of 1 tab? I will do that.
Please let me know if anything else needs improvement! Thanks.

Reply:
I converted all tabs to spaces and it is now looking fine.